staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -749,12 +757,21 @@ func (r *Request) Stream(ctx context.Context) (io.ReadCloser, error) {
 
 		default:
 			done, transformErr := func() (bool, error) {
-				defer resp.Body.Close()
+				// make sure we always drain and close
+				closer := NewResponseCloser(resp)
+				defer closer.DrainAndClose()
+
+				// get the result out of the response if provided *before* IsNextRetry,
+				// since that has to drain/close the response before resetting the request body
+				var result Result
+				if resp != nil {
+					result = r.transformResponse(resp, req)
+				}
```

Comment:
same comment about keeping transformResponse in the original location if we buffer in IsNextRetry